pkg/controllers/managementuser/clusterauthtoken/user_attribute.go

```diff
@@ -68,5 +69,13 @@ func compareUserAttributeClusterUserAttribute(userAttribute managementv3.UserAtt
 		lastRefresh:  clusterUserAttribute.LastRefresh,
 		needsRefresh: clusterUserAttribute.NeedsRefresh,
 	}
-	return groups, reflect.DeepEqual(current, old)
+	if !reflect.DeepEqual(current, old) {
+		return groups, false
+	}
+
+	if !reflect.DeepEqual(userAttribute.Extra, clusterUserAttribute.Extra) {
```

Comment:
Could we just add `Extra` to `userAttributeCompare` instead? I believe `reflect.DeepEqual` handles maps OK.